microsoft/testsuites/vm_extensions/kvvm_extension.py

```diff
@@ -0,0 +1,211 @@
+        node_capability = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+        location = node_capability.location
+
+        # User's attributes
+        user_tenant_id = os.environ.get("tenant_id")
+        if user_tenant_id is None:
+            raise ValueError("Environment variable 'tenant_id' is not set.")
+        user_object_id = os.environ.get("user_object_id")
+        if user_tenant_id is None:
+            raise ValueError("Environment variable 'user_object_id' is not set.")
+        assert user_object_id is not None
+        credential = DefaultAzureCredential(additionally_allowed_tenants=["*"])
+
+        # Identity assignment
+        compute_client = ComputeManagementClient(credential, platform.subscription_id)
+        vm = compute_client.virtual_machines.get(
+            resource_group_name, node_context.vm_name
+        )
+        object_id_vm = vm.identity.principal_id
+        if object_id_vm is None:
+            raise ValueError("object_id_vm is not set.")
+
+        # Key Vault properties
+        keyvault_client = KeyVaultManagementClient(credential, platform.subscription_id)
+        vault_properties = VaultProperties(
+            tenant_id=user_tenant_id,
+            sku=Sku(name="standard"),
+            access_policies=[
+                AccessPolicyEntry(
+                    tenant_id=user_tenant_id,
+                    object_id=user_object_id,
+                    permissions=Permissions(
+                        keys=["all"], secrets=["all"], certificates=["all"]
+                    ),
+                ),
+                AccessPolicyEntry(
+                    tenant_id=user_tenant_id,
+                    object_id=object_id_vm,
+                    permissions=Permissions(
+                        keys=["all"], secrets=["all"], certificates=["all"]
+                    ),
+                ),
+            ],
+        )
+
+        # Create Key Vault
+        parameters = VaultCreateOrUpdateParameters(
+            location=location, properties=vault_properties
+        )
+        keyvault_poller = keyvault_client.vaults.begin_create_or_update(
+            resource_group_name, vault_name_a, parameters
+        )
+        keyvault_result = keyvault_poller.result()
+
+        # Assertions and logging
+        log.info(
+            f"Provisioned vault {keyvault_result.name} "
+            f"in {keyvault_result.location} region"
+        )
+        assert_that(keyvault_result.name).described_as(
+            "Expected the Key Vault name to match the given value"
+        ).is_equal_to(vault_name_a)
+        assert_that(keyvault_result.location).described_as(
+            "Expected the Key Vault location to match the given value"
+        ).is_equal_to(location)
+
+        # Certificates
+        log.info(
+            "About to call create_certificates with vault_url: %s",
+            keyvault_result.properties.vault_uri,
+        )
```

Comment:
f"About to call create_certificates with vault_url: {keyvault_result.properties.vault_uri}"